go.mod

```diff
@@ -131,6 +133,7 @@ require (
 
 require (
 	github.com/cenkalti/backoff/v4 v4.1.2 // indirect
+	github.com/cucumber/godog v0.12.5
```

Comment:
Would it make sense to have a separate module for the e2e tests, so that these test dependencies don't become a dependency for the main module?

We took that approach in containerd, where we then replaced the main module  with a path, to make sure we use the code from the branch where it's needed ; https://github.com/containerd/containerd/blob/main/integration/client/go.mod#L79

Reply:
We can! I don't think that would be a large change, and it would allow us to separate test dependencies better. I wonder what the rest of the team thinks about that (cc @glours @ulyssessouza @milas @nicksieger).

Also whether we can/should leave splitting that out for another PR or we'd rather do it here.

I do wish Go had better tooling around that/test dependencies 😅

Reply:
That occurred to me too, we already did that for dev-envs and it seems to work well. Also would make sense in light of https://github.com/compose-spec/conformance-tests if we decide to consolidate them.

Reply:
Sounds good! I went ahead and moved the new tests into an `e2e` module, and then separately we can move the others there too :)

Reply:
Thanks! So, yes, I think the "ideal" at some point would be to (e.g) have a docker image with the integration tests compiled in, which could be run with a compose binary and docker socket mounted; this would also allow for (e.g.) https://github.com/moby/moby to run the latest e2e/integration tests as part of CI.

Reply:
Oooh, I like that idea. I'll keep that in mind for an improvement soon.